### PR TITLE
[Merged by Bors] - feat: Marica-Schönheim special case of Graham's conjecture

### PR DESCRIPTION
**Pull request by YaelDillies**
Prove the special case of Graham's conjecture when the numbers are squarefree.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/MaricaSchoenheim.lean [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+-/
+import Mathlib.Combinatorics.SetFamily.FourFunctions
+import Mathlib.Data.Nat.Squarefree
+
+/-!
+# The Marica-Schönheim special case of Graham's conjecture
+
+Graham's conjecture states that if $0 < a_1 < \dots a_n$ are integers, then
+$\max_{i, j} \frac{a_i}{\gcd(a_i, a_j)} \ge n$. This file proves the conjecture when the $a_i$ are
+squarefree as a corollary of the Marica-Schönheim inequality.
```
Comment (jcommelin) [inline Mathlib/NumberTheory/MaricaSchoenheim.lean:14]: You are calling the corollary itself `marica_schoenheim`...

Comment (YaelDillies) [reply Mathlib/NumberTheory/MaricaSchoenheim.lean:14]: Yes, as in "the Marica-Schönheim special case of Graham's conjecture". Do you have suggestions for a better name?

Comment (urkud) [reply Mathlib/NumberTheory/MaricaSchoenheim.lean:14]: `GrahamConjectureFor.of_squarefree`? Then you should define something like
```
def GrahamConjectureFor (n : ℕ) (f : Fin n ↪o ℕ) : Prop :=
  n ≠ 0 → ∃ i j, (f i).gcd (f j) * n ≤ f i
```

Comment (YaelDillies) [reply Mathlib/NumberTheory/MaricaSchoenheim.lean:14]: Hmm... maybe, but Graham's conjecture has been proved in the general case. Proof is hard though. Here I'm just trying to demonstrate the use of the Marica-Schönheim inequality.

Comment (urkud) [reply Mathlib/NumberTheory/MaricaSchoenheim.lean:14]: Then this `def` would be a good place to explain that the general proof is available but has not been formalized yet.

Comment (YaelDillies) [reply Mathlib/NumberTheory/MaricaSchoenheim.lean:14]: What about now?

File: Mathlib/NumberTheory/MaricaSchoenheim.lean [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+open scoped BigOperators FinsetFamily
+
+namespace Nat
+
+/-- Statement of Graham's conjecture (which is now a theorem in the literature).
+
+Graham's conjecture states that if $0 < a_1 < \dots a_n$ are integers, then
+$\max_{i, j} \frac{a_i}{\gcd(a_i, a_j)} \ge n$. -/
+def GrahamConjecture (n : ℕ) (f : ℕ → ℕ) : Prop :=
+  ∃ i < n, ∃ j < n, (f i).gcd (f j) * n ≤ f i
```
Comment (urkud) [inline Mathlib/NumberTheory/MaricaSchoenheim.lean:31]: I would prefer `GrahamConjecture` to be true, i.e., include assumptions on `n` and `f`.

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**